control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -316,6 +316,9 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		if shouldCleanupCloudResources(r.Log, hostedControlPlane) {
 			if err := r.destroyAWSDefaultSecurityGroup(ctx, hostedControlPlane); err != nil {
 				condition.Message = "failed to delete AWS default security group"
+				if awsErrorCode(err) == "DependencyViolation" {
```

Comment:
maybe make the `destroyAWSDefaultSecurityGroup`  function return the aws error code, so we don't have to parse multiple times

Reply:
that function returns multiple other formatted errors and I am not sure if the AWS errors  would be as specific.

Reply:
I meant in addition to the error